cmd/config/config_yaml.go

```diff
@@ -634,11 +633,6 @@ func (c YAMLConfig) parseFilterConfig() *filter.Config {
 }
 
 func (c TransformationsConfig) parseTransformationConfig() (*transformer.Config, error) {
-	if len(c.TransformerRules) == 0 {
```

Comment:
If there are no transformation rules we can skip the transformer layer altogether, so we should return `nil,nil` which should cover both yaml and env configs (since this method is used by both).

Reply:
We can check if the rules are provided (which can be mandatory), by checking if the c.TransformerRules is `nil`, and return the error we were returning until now. And then do a check if the length is 0, in which case we just skip the transformation layer altogether.